source/local/local.go

```diff
@@ -191,8 +191,12 @@ func (ls *localSourceHandler) snapshot(ctx context.Context, s session.Group, cal
 			}
 			stat.Uid = uint32(identity.UID)
 			stat.Gid = uint32(identity.GID)
-			return true
 		}
+		// whatever permissions the user has, give them to group and others as well
+		// this matches behavior of gitsource, given that umask is 0
+		umode := (stat.Mode & 0700) >> 6
+		stat.Mode = (stat.Mode ^ (stat.Mode & 0777)) | umode | (umode << 3) | (umode << 6)
```

Comment:
can you explain/comment more what this does exactly. Should this change be in https://github.com/tonistiigi/fsutil ?

Reply:
Done - added more info about the bitwise operations. Not sure if that's what you meant. The more extended explanation is given in the PR description - I can include some of that too if you think that's useful.

Don't think this should be in fsutil, because there are valid uses of fsutil where the user might want to preserve mode for *group* and *other*.